pkg/server/httpserver/auth.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpserver
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+
+	"github.com/chaos-mesh/chaosd/pkg/server/utils"
+)
+
+const (
+	MTLSServer = "mTLS"
+	TLSServer  = "tls"
```

Comment:
In which case should we use `TLSServer`/`MTLSServer`?

Reply:
I probably should add a comment there :thinking:. But to summarize,
1. **TLS:** Standard SSL encryption on requests with the provided server certificate and key. Basically clients will do HTTPS request without presenting any certificates.
2. **mTLS (or Mutual TLS):** Two-way SSL where both server and client presents their certificates. Basically clients do HTTPS request, validates server certificate and then presents its own client certificate. If the client certificate is valid (if it is signed by the CA certificate provided to chaosd), then the client is authenticated and allowed to call any API endpoint.

Reply:
OK, got it